crates/llms/src/xai.rs

```diff
@@ -98,8 +99,36 @@ impl Xai {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+struct Model {
+    id: String,
+    created: u64,
+    object: String,
+    owned_by: String,
+}
+
 #[async_trait]
 impl Chat for Xai {
+    async fn health(&self) -> Result<(), Error> {
+        let span = tracing::span!(target: "task_history", tracing::Level::INFO, "health", input = "health");
+        match self
+            .client
+            .get::<Model>(format!("/models/{}", self.model).as_str())
+            .await
+        {
+            Ok(_) => Ok(()),
+            Err(e) => {
+                tracing::error!(target: "task_history", parent: &span, "{e}");
+                Err(Error::HealthCheckError {
+                    source: Box::<dyn std::error::Error + Send + Sync>::from(format!(
+                        "Model {} not found",
+                        self.model
+                    )),
+                })
```

Comment:
```suggestion
                Err(Error::ModelNotFound {
                    model: self.model,
                    model_source: "xai".to_string()
                })
```

This could use a dedicated error enum like:

```
#[snafu(display("The specified model identifier '{model}' is not valid for the source '{model_source}'.\nVerify the model exists, and try again."))]
ModelNotFound { model: String, model_source: String },
```